project/src/main/java/com/google/sps/servlets/EventCreationServlet.java

```diff
@@ -93,10 +94,13 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
 
     User host = SpannerTasks.getLoggedInUser().get();
     Event event = new Event.Builder(name, description, labels, location, date, time, host).build();
-    SpannerTasks.insertorUpdateEvent(event);
-
-    // Add event to search index
-    searchStore.addEventToIndex(event.getId(), name, description);
+    if (eventId == null) {
+      SpannerTasks.insertorUpdateEvent(event);
+      searchStore.addEventToIndex(event.getId(), name, description);
+    } else {
+      event = event.toBuilder().setId(eventId).build();
+      SpannerTasks.insertorUpdateEvent(event);
```

Comment:
i would just inline

Reply:
I tried this but on line 105 you have a line that gets the event ID of the event, so we should not inline.